source/gui/addonGui.py

```diff
@@ -1,818 +1,817 @@
+			"Would you still like to update?"
+		).format(summary=summary, newVersion=newVersion)
+
+		updateAddonInstallationMessage = _(
+			# Translators: A message asking if the user wishes to update a previously installed
+			# add-on with this one.
+			"A version of this add-on is already installed. "
+			"Would you like to update {summary} version {curVersion} to version {newVersion}?"
+		).format(summary=summary, curVersion=curVersion, newVersion=newVersion)
+
+		if gui.messageBox(
+			overwriteExistingAddonInstallationMessage if curVersion == newVersion else updateAddonInstallationMessage,
+			messageBoxTitle,
+			wx.YES|wx.NO|wx.ICON_WARNING
+		) != wx.YES:
+				return False
+
+	from contextlib import contextmanager
+
+	@contextmanager
+	def doneAndDestroy(window):
+		try:
+			yield window
+		except:
+			# pass on any exceptions
+			raise
+		finally:
+			# but ensure that done and Destroy are called.
+			window.done()
+			window.Destroy()
+
+	#  use a progress dialog so users know that something is happening.
+	progressDialog = gui.IndeterminateProgressDialog(
+		parentWindow,
+		# Translators: The title of the dialog presented while an Addon is being installed.
+		_("Installing Add-on"),
+		# Translators: The message displayed while an addon is being installed.
+		_("Please wait while the add-on is being installed.")
+	)
+
+	try:
+		# Use context manager to ensure that `done` and `Destroy` are called on the progress dialog afterwards
+		with doneAndDestroy(progressDialog):
+			gui.ExecAndPump(addonHandler.installAddonBundle, bundle)
+			if prevAddon:
+				prevAddon.requestRemove()
+			return True
+	except:
+		log.error("Error installing  addon bundle from %s" % addonPath, exc_info=True)
+		gui.messageBox(
+			# Translators: The message displayed when an error occurs when installing an add-on package.
+			_("Failed to install add-on from %s") % addonPath,
+			# Translators: The title of a dialog presented when an error occurs.
+			_("Error"),
+			wx.OK | wx.ICON_ERROR
+		)
+	return False
+
+
+def handleRemoteAddonInstall(addonPath):
+	# Add-ons cannot be installed into a Windows store version of NVDA
+	if config.isAppX:
+		gui.messageBox(
+			# Translators: The message displayed when an add-on cannot be installed due to NVDA running as a Windows Store app
+			_("Add-ons cannot be installed in the Windows Store version of NVDA"),
+			# Translators: The title of a dialog presented when an error occurs.
+			_("Error"),
+			wx.OK | wx.ICON_ERROR)
+		return
+	gui.mainFrame.prePopup()
+	if installAddon(gui.mainFrame, addonPath):
+		promptUserForRestart()
+	gui.mainFrame.postPopup()
+
+
+def _showAddonRequiresNVDAUpdateDialog(parent, bundle):
+	incompatibleReasonMessage = _(
+		# Translators: The message displayed when installing an add-on package is prohibited,
+		# because it requires a newer version of NVDA than is currently in use.
+		"Installation of this add-on has been blocked. Your version of NVDA ({NVDAVersion}) "
```

Comment:
I think "failed" is more typical UX terminology than "blocked"

```suggestion
		"Installation of this add-on has failed. Your version of NVDA ({NVDAVersion}) "
```

Reply:
I don't mind that change if you really want it, but I think "blocked" is more accurate.

"failed" implies that some error condition or resolvable problem keeps the add-on from being installed.
Here, however, this is NVDA intentionally refusing to install this add-on, even though technically it could be installed.

In fact, other add-ons with these manifest conditions that were already installed when NVDA was updated, are not uninstalled. They are simply disabled. So there is no technical reason at all why an add-on could not be installed and kept disabled under these circumstances, except that NVDA is, quite literally, blocking it from happening.

As an aside, note that the "has been blocked" language comes from the original versions of these dialogs. That is the one part of the messages that I didn't change. :)

Reply:
I'm not particularly fussed. I'd like to leave this PR open for a short while to see what others think.